adapters/bigoad/bigoad.go

```diff
@@ -0,0 +1,156 @@
+package bigoad
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"text/template"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v2/adapters"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/errortypes"
+	"github.com/prebid/prebid-server/v2/macros"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint *template.Template
+}
+
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	template, err := template.New("endpointTemplate").Parse(config.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse endpoint url template: %v", err)
+	}
+
+	bidder := &adapter{
+		endpoint: template,
+	}
+
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	bigoadExt, err := getImpExt(&request.Imp[0])
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	resolvedUrl, err := a.buildEndpointURL(bigoadExt)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	requestJSON, err := json.Marshal(request)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	requestData := &adapters.RequestData{
+		Method:  "POST",
+		Uri:     resolvedUrl,
+		Body:    requestJSON,
+		Headers: getHeaders(request),
+		ImpIDs:  openrtb_ext.GetImpIDs(request.Imp),
+	}
+
+	return []*adapters.RequestData{requestData}, nil
+}
+
+func getHeaders(request *openrtb2.BidRequest) http.Header {
+	headers := http.Header{}
+	addNonEmptyHeaders(&headers, map[string]string{
+		"Content-Type":      "application/json;charset=utf-8",
+		"Accept":            "application/json",
+		"x-openrtb-version": "2.5",
+	})
+	return headers
+}
+
+func addNonEmptyHeaders(headers *http.Header, headerValues map[string]string) {
+	for key, value := range headerValues {
+		if len(value) > 0 {
+			headers.Add(key, value)
+		}
+	}
+}
+
+func getImpExt(imp *openrtb2.Imp) (*openrtb_ext.ExtImpBigoAd, error) {
+	var bidderExt adapters.ExtImpBidder
+	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		return nil, &errortypes.BadInput{
+			Message: fmt.Sprintf("imp %s: unable to unmarshal ext", imp.ID),
+		}
+	}
+	var bigoadExt openrtb_ext.ExtImpBigoAd
+	if err := json.Unmarshal(bidderExt.Bidder, &bigoadExt); err != nil {
+		return nil, &errortypes.BadInput{
+			Message: fmt.Sprintf("imp %s: unable to unmarshal ext.bidder: %v", imp.ID, err),
+		}
+	}
+	imp.Ext = bidderExt.Bidder
+	return &bigoadExt, nil
+}
+
+func (a *adapter) buildEndpointURL(params *openrtb_ext.ExtImpBigoAd) (string, error) {
+	endpointParams := macros.EndpointTemplateParams{Host: params.Host, SspId: params.SspId}
+	return macros.ResolveMacros(a.endpoint, endpointParams)
```

Comment:
Host is not needed in ` https://api.imotech.tech/Ad/GetAdOut?sspid={{.SspId}}` bidder endpoint

should remove `Host` from `macros.EndpointTemplateParams`